client/my-sites/stats/site.jsx

```diff
@@ -914,6 +916,11 @@ const StatsSite = ( props ) => {
 		[]
 	); // Track the last viewed tab.
 
+	useEffect( () => {
+		const query = context.query;
+		recordCurrentScreen( 'traffic', query, true );
+	}, [ periodProp, context.query ] );
```

Comment:
The `period` change is detected but not recorded in the history. It also seems reasonable to store the `period` property for a more accurate history, doesn't it?

Reply:
You are right about `periodProp`, and I have removed this. Re: period, I have noticed the page is redirected to the correct period `/month` / `/day` etc based on the query parameters so I decided to skip that logic from here. So right now the back link for traffic page is always `/day` with the query parameters and that redirects to correct path. Do you think this would be Ok or we should also store and redirect to the correct period too?

Reply:
I actually went ahead and pushed a fix to handle the period because

1. This will avoid URL redirect/change when the page is loading
2. This is not handled for other screens like `/subscribers`, so fixing this here will help maintain the period in URL for other screens